// @bit/nexxtway.react-rainbow.types/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BaseProps", {
  enumerable: true,
  get: function get() {
    return _types.BaseProps;
  }
});
Object.defineProperty(exports, "AvatarSize", {
  enumerable: true,
  get: function get() {
    return _types.AvatarSize;
  }
});
Object.defineProperty(exports, "MenuSize", {
  enumerable: true,
  get: function get() {
    return _types.MenuSize;
  }
});
Object.defineProperty(exports, "MenuAlignment", {
  enumerable: true,
  get: function get() {
    return _types.MenuAlignment;
  }
});
Object.defineProperty(exports, "ButtonType", {
  enumerable: true,
  get: function get() {
    return _types.ButtonType;
  }
});
Object.defineProperty(exports, "ButtonIconSize", {
  enumerable: true,
  get: function get() {
    return _types.ButtonIconSize;
  }
});
Object.defineProperty(exports, "ButtonIconVariant", {
  enumerable: true,
  get: function get() {
    return _types.ButtonIconVariant;
  }
});
Object.defineProperty(exports, "IconPosition", {
  enumerable: true,
  get: function get() {
    return _types.IconPosition;
  }
});
Object.defineProperty(exports, "LookupValue", {
  enumerable: true,
  get: function get() {
    return _types.LookupValue;
  }
});
Object.defineProperty(exports, "RadioOption", {
  enumerable: true,
  get: function get() {
    return _types.RadioOption;
  }
});

var _types = require("./types");
